pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -356,11 +357,50 @@ func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, d
 		}
 
 		result = append(result, machineDeployment)
+
+		// if a health check timeout was specified create health checks for this machine pool
+		if machinePool.UnhealthyNodeTimeout != nil {
+			hc := deploymentHealthChecks(machineDeployment, machinePool.NodeStartupTimeout, machinePool.UnhealthyNodeTimeout, machinePool.MaxUnhealthy, machinePool.UnhealthyRange)
```

Comment:
nit: just pass the `machinePool` to this function to make the signature simpler.